### ISSUE TITLE
Cannot redeclare isHex()

### ISSUE THREAD

**Issue by tfischer77** [nilsteampassnet/TeamPass#4424]
### Steps to reproduce
1. Install or upgrade Teampass to the latest Master
2. Try to login for the first time

### Expected behaviour
Login should be possible

### Actual behaviour
Login is not possible, instead, sandbox is rotating forever.
In Apache error log, the following appears:
[Thu Oct 17 09:26:09.998169 2024] [php:error] [pid 480935:tid 480935] [client xx.xx.xx.xx:48504] PHP Fatal error:  Cannot redeclare isHex() (previously declared in /var/www/html/TeamPass-master/sources/main.functions.php:104) in /var/www/html/teampass/sources/main.functions.php on line 104, referer: https://hostname.com/teampass/index.php

### Server configuration
**Operating system**:
Debian 12.7
**Web server:**
Apache 2.4.62
**Database:**
10.11.6-MariaDB-0+deb12u1
**PHP version:**
PHP Version 8.2.20
**Teampass version:** 
Master from Github (Oct 16, 2024)
**Teampass configuration file:**
// DATABASE connexion parameters
define("DB_HOST", "localhost:3306");
define("DB_USER", "teampass");
define("DB_PASSWD", "XXXXX");
define("DB_NAME", "teampass");
define("DB_PREFIX", "teampass_");
define("DB_PORT", "3306");
define("DB_ENCODING", "utf8");
//define("DB_SSL", false); // if DB over SSL then comment this line
// if DB over SSL then uncomment the following lines
define("DB_SSL", array(
    "key" => "",
    "cert" => "",
    "ca_cert" => "",
    "ca_path" => "",
    "cipher" => ""
));
define("DB_CONNECT_OPTIONS", array(
    MYSQLI_OPT_CONNECT_TIMEOUT => 10
));
define("SECUREPATH", "/var/local/teampass");
define("SECUREFILE", "teampass-seckey.txt");
define("IKEY", "");
define("SKEY", "");
define("HOST", "");

if (isset($_SESSION['settings']['timezone']) === true) {
    date_default_timezone_set($_SESSION['settings']['timezone']);
}

### Client configuration
**Browser:**
Google Chrome 129.0.6668.101
**Operating system:**
Windows 10
#### Web server error log
```
[Thu Oct 17 09:26:09.998169 2024] [php:error] [pid 480935:tid 480935] [client xx.xx.xx.xx:48504] PHP Fatal error:  Cannot redeclare isHex() (previously declared in /var/www/html/TeamPass-master/sources/main.functions.php:104) in /var/www/html/teampass/sources/main.functions.php on line 104, referer: https://<host>.com/teampass/index.php

```

#### Log from the web-browser developer console (CTRL + SHIFT + i)
I didn't catch the console error message exactly, but it stated something like "Error 500" with identity

### Possible solution ###
I replaced all occurences of "include_once" in the source code by "require_once". Issue was immediately solved and login was possible.
```
find . -type f -name "*.php" -print0 | xargs -0 sed -i '' -e 's/include_once/require_once/g'
```
I already downloaded the master tree two weeks ago. There, I couldn't find this issue. What did change on the sources?
Best regards
Tim

**Comment by nilsteampassnet** [comment]
@tfischer77 
Strange error as this fucntion is only declared once.

> PHP Fatal error: Cannot redeclare isHex() (previously declared in /var/www/html/TeamPass-master/sources/main.functions.php:104) in /var/www/html/teampass/sources/main.functions.php on line 104,

The error itself points the same line as if it was "auto-calling".

THe solution is also quite strange. require_once generates an error if missing while include_once raises only a warning ... how can this change resolve the initial error.

Did you removed all previous files before upgrading to newest master commit?

**Comment by tfischer77** [comment]
I reinstalled the actual master now on a newly created Linux virtual machine. Login is working without any issues.
What I did last time: I moved the database to a new server and performed an upgrade. I noticed that the Teampass location (URL) was still pointing to the old server (which was also accessible from the new server). Could that lead to the issue described?
Anyhow, as things seem to run, I'm OK to close the issue.
